consumer/consumer_test.go

```diff
@@ -47,3 +49,164 @@ func TestConsumerConstructorNoKafka(t *testing.T) {
 		t.Fatal("consumer.New should return nil instead of Consumer implementation")
 	}
 }
+
+func TestParseEmptyMessage(t *testing.T) {
+	const message = ``
+	_, _, _, err := consumer.ParseMessage([]byte(message))
+	if err == nil {
+		t.Fatal("Error is expected to be returned for empty message")
+	}
+}
+
+func TestParseMessageWithWrongContent(t *testing.T) {
+	const message = `{"this":"is", "not":"expected content"}`
+	_, _, _, err := consumer.ParseMessage([]byte(message))
+	if err == nil {
+		t.Fatal("Error is expected to be returned for empty message")
+	}
+}
+
+func TestParseMessageWithImproperJSON(t *testing.T) {
+	const message = `"this_is_not_json_dude"`
+	_, _, _, err := consumer.ParseMessage([]byte(message))
+	if err == nil {
+		t.Fatal("Error is expected to be returned for empty message")
+	}
+}
```

Comment:
Can the `err` be checked in order to see if the error is the expected one?

Reply:
you are right, I'll fix it.